sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosResponseDiagnosticsTest.java

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class CosmosResponseDiagnosticsTest extends TestSuiteBase {
+    private CosmosClient gatewayClient;
+    private CosmosClient directClient;
+    private CosmosContainer container;
+    private CosmosAsyncContainer cosmosAsyncContainer;
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() throws Exception {
+        assertThat(this.gatewayClient).isNull();
+        CosmosClientBuilder cosmosClientBuilder = new CosmosClientBuilder()
+            .setEndpoint(TestConfigurations.HOST)
+            .setKey(TestConfigurations.MASTER_KEY);
+        ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+        connectionPolicy.setConnectionMode(ConnectionMode.GATEWAY);
+        gatewayClient = cosmosClientBuilder.setConnectionPolicy(connectionPolicy).buildClient();
+        connectionPolicy = new ConnectionPolicy();
+        connectionPolicy.setConnectionMode(ConnectionMode.DIRECT);
+        directClient = cosmosClientBuilder.setConnectionPolicy(connectionPolicy).buildClient();
+        cosmosAsyncContainer = getSharedMultiPartitionCosmosContainer(this.gatewayClient.asyncClient());
+        container = gatewayClient.getDatabase(cosmosAsyncContainer.getDatabase().getId()).getContainer(cosmosAsyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() throws CosmosClientException {
+        assertThat(this.gatewayClient).isNotNull();
+        this.gatewayClient.close();
+        if (this.directClient != null) {
+            this.directClient.close();
+        }
+    }
+
+    @Test(groups = {"simple"})
+    public void gatewayDiagnostics() throws CosmosClientException {
+        CosmosItemProperties cosmosItemProperties = getCosmosItemProperties();
+        CosmosItemResponse createResponse = container.createItem(cosmosItemProperties);
+        String diagnostics = createResponse.getCosmosResponseDiagnostics().toString();
+        assertThat(diagnostics).contains("Connection Mode : " + ConnectionMode.GATEWAY);
+        assertThat(diagnostics).contains("Gateway statistics");
+        assertThat(diagnostics).contains("Operation Type : " + OperationType.Create);
+        assertThat(createResponse.getCosmosResponseDiagnostics().getRequestLatency()).isNotNull();
+        deleteItem(createResponse);
```

Comment:
as long as you are using a shared collection from TestSuiteBase, you don't need to delete documents, the test framework will do that when needed.

please remove deleteItem here and elsewhere

Reply:
done